lua/autorun/nl_library.lua

```diff
@@ -0,0 +1,476 @@
+		if IsValid(v.nllragdoll) then
+			//Send new health. The normal health sending is somehow broken when ragdolled.
+			if v:Health() != v.lasthp then
+				net.Start("nllsendhealth")
+					net.WriteEntity(v)
+					net.WriteInt(v:Health(),32)
+				net.Broadcast()
+				v.lasthp = v:Health()
+			end
+			
+			local rag = v.nllragdoll
+			local phys = rag:GetPhysicsObjectNum(0)
+			if phys:IsValid() and v.condition == "STN" then // This player is painfully stunned, not asleep, shake them a LITTLE bit.
+				phys:AddAngleVelocity(Vector(0,math.sin(CurTime() * 2) * 100,0))
+			end
+		end
+	end
+end)
+
+//Stores the players weaponclasses and ammo in a table.
+function NLL.PlyStoreWeapons(ply)
+	ply.storeweps = {}
+	for k,v in pairs(ply:GetWeapons()) do
+		table.insert(ply.storeweps, {cl = v:GetClass(), c1 = v:Clip1(), c2 = v:Clip2()})
+	end
+end
+
+//Retrieves the stored weapons
+function NLL.PlyRetrieveWeapons(ply)
+	for k,v in pairs(ply.storeweps or {}) do
+		ply:Give(v.cl)
+		local wep = ply:GetWeapon(v.cl)
+		if IsValid(wep) then
+			wep:SetClip1(v.c1)
+			wep:SetClip2(v.c2)
+		end
+	end
+end
+
+/*
+Makes a hull trace the size of a player.
+*/
+local data = {}
+function NLL.PlayerHullTrace(pos, ply, filter)
+	data.start = pos
+	data.endpos = pos
+	data.filter = filter
+	
+	return util.TraceEntity( data, ply )
+end
+
+/*
+Attemps to place the player at this position or as close as possible.
+*/
+// Directions to check
+local directions = {
+	Vector(0,0,0), Vector(0,0,1), //Center and up
+	Vector(1,0,0), Vector(-1,0,0), Vector(0,1,0), Vector(0,-1,0) //All cardinals
+}
+
+for deg=45,315,90 do //Diagonals
+	local r = math.rad(deg)
+	table.insert(directions, Vector(math.Round(math.cos(r)), math.Round(math.sin(r)), 0))
+end
+
+local magn = 15 // How much increment for each iteration
+local iterations = 2 // How many iterations
+function NLL.PlayerSetPosNoBlock( ply, pos, filter )
+	local tr
+	
+	local dirvec
+	local m = magn
+	local i = 1
+	local its = 1
+	repeat
+		dirvec = directions[i] * m
+		i = i + 1
+		if i > #directions then
+			its = its + 1
+			i = 1
+			m = m + magn
+			if its > iterations then
+				ply:SetPos(pos) // We've done as many checks as we wanted, lets just force him to get stuck then.
+				return false
+			end
+		end
+		
+		tr = NLL.PlayerHullTrace(dirvec + pos, ply, filter)
+	until tr.Hit == false
+	
+	ply:SetPos(pos + dirvec)
+	return true
+end
+
+
+/*
+Sets the player invisible/visible
+*/
+function NLL.PlayerInvis( ply, bool )
+	ply:SetNoDraw(bool)
+	ply:DrawShadow(not bool)
+
+	if pac and pac.TogglePartDrawing then
+		pac.TogglePartDrawing(ply, not bool)
+	end
+	/*
+	ply:SetMaterial( bool and "models/effects/vol_light001" or "" )
+	ply:SetRenderMode( bool and RENDERMODE_TRANSALPHA or RENDERMODE_NORMAL )
+	ply:Fire( "alpha", bool and 0 or 255, 0 )
+	*/
+end
+
+
+/*
+Deploy player ragdoll, Sleeping Variants can be woken up with E.
+*/
+function NLL.Ragdoll( ply, pushdir, stuntype)
+	if not SERVER then return end
```

Comment:
If this function is only meant for server, then only define it there using `if SERVER then`